packages/core/src/Utils/Language/language.js

```diff
@@ -9,7 +9,10 @@ export const currentLanguage = getLanguage();
 
 export const getURL = lang => urlForLanguage(lang);
 
-export const changeLanguage = (key, changeCurrentLanguage) => {
+let timerID;
+export const changeLanguage = (key, changeCurrentLanguage, setLanguageChanging) => {
+    if (timerID) clearTimeout(timerID);
```

Comment:
added clear interval for such cases, when user decided to change language several times in a row. So it prevents unexpected call setLanguageChanging in previous setTimeout